xdsl/parser.py

```diff
@@ -276,9 +276,7 @@ def _find_token_end(self, start: int | None = None) -> int:
                 lambda x: x >= 0,
                 (self.input.content.find(part, i) for part in self._break_on),
             ))
-        # Make sure that we break at some point
-        break_pos.append(self.input.len)
-        return min(break_pos)
+        return min(min(break_pos), self.input.len)
```

Comment:
The one risk with this that I now realise is that `min(break_pos)` might be None, so actually this might be better with
```suggestion
        min_break_pos = min(break_pos
        return self.input.len if min_break_pos is None else min_break_pos
```

Reply:
No, this doesn't work, because `min(break_pos)` is raising an exception if `break_pos` is empty.
I would say that we are going to bikeshedding territory here though, it doesn't really matter if we create a list to use it only once right? Also, note that this code is going to be deleted once my lexer replace the tokenizer.

Reply:
yeah I mean it's all fairly minimal. My original comment was more as a general convention to reach for list comprehensions rather than filter/map etc as they work better and are more readable if used consistently